Doc/library/io.rst

```diff
@@ -64,6 +64,12 @@ In-memory text streams are also available as :class:`StringIO` objects::
 
    f = io.StringIO("some initial text data")
 
+.. note::
+   When working with a non-blocking stream, be aware that operations on text I/O objects
```

Comment:
"reads" or "read operations" rather than "operations".   and change the later text to "if the stream cannot perform the operation immediately."